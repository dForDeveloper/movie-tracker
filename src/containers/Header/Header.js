import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { setUser, setFavorites } from '../../actions';

export class Header extends Component {
  handleClick = () => {
    this.props.setUser({});
    this.props.setFavorites([]);
  }

  getUserLinks = () => {
    const { currentUser } = this.props
    if (currentUser.name) {
      return (
        <div className="user-links">
          <p>Hello, { currentUser.name }</p>
          <Link to='/favorites' className="favorites-link">View Favorites</Link>
          <Link to='/' className="logout-link" onClick={this.handleClick}>Log Out</Link>
        </div>
      );
    } else {
      return (
        <div className="user-links">
          <Link to='/sign-up'>Sign Up</Link>
          <Link to='/login'>Log In</Link>
          <p>Log in to view your favorited movies.</p>
        </div>
      );
    }
  }

  render() {
    return (
      <header className="Header">
        {this.getUserLinks()}
      </header>
    );
  }
}

export const mapStateToProps = (state) => ({
  currentUser: state.currentUser
});

export const mapDispatchToProps = (dispatch) => ({
  setUser: () => dispatch(setUser()),
  setFavorites: () => dispatch(setFavorites())
}); 

export default connect(mapStateToProps, mapDispatchToProps)(Header)

Header.propTypes = {
  currentUser: PropTypes.object,
  setFavorites: PropTypes.func,
  setUser: PropTypes.func
};