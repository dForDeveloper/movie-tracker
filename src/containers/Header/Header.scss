.Header {
  grid-column: 1 / 5;
  width: 100%;
  position: sticky;
  top: 0;
  background-color: #808daf;
  background-image: linear-gradient(to right, #b3afcc, $steel-grey, white 95%);
}

.user-links {
  display: flex;
  align-items: center;
  padding: 0 20px;
  margin: auto;
  font-family: $body-font;
  font-size: 20px;
  color: $text-color;
  border-top: 4px solid black;
  border-bottom: 2px solid black;
}

.nav-left {
	display: flex;
  justify-content: space-around;
}

.nav-right {
  margin-left: auto;
}

#item-left,
#item-right {
  padding: 10px;
  color: $off-white;
}

#logout-link {
  padding: 10px;
  color: black;
}

.greeting {
  padding: 10px;
}

.username {
  padding: 0px 5px;
  font-weight: bold;
}

#item-left:hover,
#item-right:hover,
.logout-link:hover {
  background-color: white;
  color: $text-color;
}

#logout-link:hover {
  background-color: $text-color;
  color: $off-white;
}
