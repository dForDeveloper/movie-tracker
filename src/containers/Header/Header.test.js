import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

const setUserMock = jest.fn();
const setFavoritesMock = jest.fn();
let mockCurrentUser = { name: 'shannon' }

describe('Header', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Header 
        setUser={setUserMock} 
        setFavorites={setFavoritesMock} 
        currentUser={mockCurrentUser} 
      />
    );
  });

  describe('Header container', () => {
    it('should match the snapshot when the user is not logged in', () => {
      expect(wrapper).toMatchSnapshot();
    });
    
    it('should call setUser and setFavorites when handleClick is invoked', () => {
      wrapper.find('.logout-link').simulate('click');
      expect(setUserMock).toBeCalled();
      expect(setFavoritesMock).toBeCalled();
    });

    it('should match the snapshot when the user is not logged in', () => {
      mockCurrentUser = {};
      wrapper = shallow(
        <Header 
          setUser={setUserMock} 
          setFavorites={setFavoritesMock} 
          currentUser={mockCurrentUser} 
        />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });



  describe('mapStateToProps', () => {});

  describe('mapDispatchToProps', () => {});

});