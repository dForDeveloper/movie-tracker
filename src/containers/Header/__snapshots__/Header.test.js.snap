// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header container should match the snapshot when the user is not logged in 1`] = `
<header
  className="Header"
>
  <div
    className="user-links"
  >
    <p>
      Hello, 
      shannon
    </p>
    <Link
      className="favorites-link"
      replace={false}
      to="/favorites"
    >
      View Favorites
    </Link>
    <Link
      className="logout-link"
      onClick={[Function]}
      replace={false}
      to="/"
    >
      Log Out
    </Link>
  </div>
</header>
`;

exports[`Header Header container should match the snapshot when the user is not logged in 2`] = `
<header
  className="Header"
>
  <div
    className="user-links"
  >
    <Link
      replace={false}
      to="/sign-up"
    >
      Sign Up
    </Link>
    <Link
      replace={false}
      to="/login"
    >
      Log In
    </Link>
    <p>
      Log in to view your favorited movies.
    </p>
  </div>
</header>
`;
